### PR TITLE
Projection expressions

### LINKED ISSUE #5039: Support Projection Expressions
As a user I'd like to assign a projection to discrete zoom levels

### PR DESCRIPTION
**Pull request by birkskyum**
Blocked by / How to use
- https://github.com/maplibre/maplibre-style-spec/pull/888
- https://github.com/maplibre/maplibre-style-spec/pull/899
- #5020 

Closes #5039 
- #5039 

`interpolate` example

```ts
const exprestInt = createExpression(['interpolate', ['linear'], ['zoom'], 10, 'vertical-perspective', 12, 'mercator'], v8.projection.type)
```

https://github.com/user-attachments/assets/4382040a-eacb-4b60-abba-188d10cbfba0

`step` example
```ts
const exprestInt = createExpression(['step', ['zoom'], 'vertical-perspective', 12, 'mercator'], v8.projection.type)
```

https://github.com/user-attachments/assets/1a8b71b1-6e84-4b6a-8ffd-96bd67eeab5a

## Launch Checklist

 - [X] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [X] Add an entry to `CHANGELOG.md` under the `## main` section.

### CODE REVIEW

File: src/ui/events.ts [hunk 1]
```diff
@@ -752,7 +752,7 @@ export type MapProjectionEvent = {
      * Specifies the name of the new projection.
      * Additionally includes 'globe-mercator' to describe globe that has internally switched to mercator.
      */
-    newProjection: ProjectionSpecification['type'] | 'globe-mercator';
+    newProjection: ProjectionConfigSpecification['type'] | 'globe-mercator';
```
Comment (HarelM) [inline src/ui/events.ts:755]: Remove `globe-mercator`?

**Comment by HarelM** [comment]
I was expecting the code here to use the expression value of projection, is it using the expression value somewhere that I have missed?

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 ### ✨ Features and improvements
 - Support Vertical Perspective projection ([#5023](https://github.com/maplibre/maplibre-gl-js/pull/5023))
+- Support projection expression support.
```
Comment (HarelM) [inline CHANGELOG.md:5]: Add link to PR.

**Comment by birkskyum** [comment]
@HarelM , there is some preparation work needed in GL JS before this can be hooked up to the map style.

One way to resolve it could be to make this PR just a "'update style-spec", so that we can refactor this in steps on the main branch, until we can hook it up to the map.style, and _then_  make the changelog item for the new feature

For example, I see some issues with tile loading, when going between projections:
https://github.com/user-attachments/assets/8e419442-388c-4e40-8ab2-4d3f44bd1166

And if terrain is enabled, the globe fog appear to break things
https://github.com/user-attachments/assets/9c6e8dbe-3d3e-41f3-bc47-c00e355d1a6d

**Comment by HarelM** [comment]
Spec package will be updated automatically in a few days by dependabot.
If there are other changes related to this package update then it should be part of this PR I guess. 
Other existing bugs not related to the sec change should be handled regardless of this change.

**Comment by birkskyum** [comment]
Great, I'll leave it to dependabot - wasn't sure if it handled major-release updates too

**Pull request closed without merging**